src/lib/y2network/sysconfig_routes_file.rb

```diff
@@ -0,0 +1,194 @@
+    # and keeps just <ip> part in "destination" and puts "/<prefix_len>" into
+    # "netmask"
+    #
+    # @param routes [Array<Hash>] in quad or CIDR flavors (see {#Routes})
+    # @return [Array<Hash>] in quad or slash flavor
+    def normalize_entries(entries)
+      return entries if entries.nil? || entries.empty?
+
+      entries.map do |entry|
+        subnet, prefix = entry["destination"].split("/")
+
+        next entry if prefix.nil?
+
+        entry["destination"] = subnet
+        entry["netmask"] = "/#{prefix}"
+
+        entry
+      end
+    end
+
+    # Given an IP and a netmask, returns a valid IPAddr object
+    #
+    # @param ip_str      [String] IP address; {MISSING_VALUE} means that the IP is not defined
+    # @param netmask_str [String] Netmask; {MISSING_VALUE} means than no netmask was specified
+    # @return [IPAddr,nil] The IP address or `nil` if the IP is missing
+    def build_ip(ip_str, netmask_str = MISSING_VALUE)
+      return nil if ip_str == MISSING_VALUE
+
+      ip = IPAddr.new(ip_str)
+      netmask_str == MISSING_VALUE ? ip : ip.mask(netmask_str)
+    end
+
+    # Build a route given a hash from the SCR agent
+    #
+    # @param hash [Hash] Hash from the `.routes` SCR agent
+    # @return Route
+    def build_route(hash)
+      # TODO: check whether the iface is configured in the system
+      iface = Interface.new(hash["device"])
+      # normalized SCR output contains either subnet mask or /<prefix length> under
+      # "netmask" key
+      # TODO: this should be improved in normalize_routes
+      mask = hash["netmask"] =~ /\/[0-9]+/ ? hash["netmask"][1..-1] : hash["netmask"]
+
+      Y2Network::Route.new(
+        to:        hash["destination"] != DEFAULT_DEST ? build_ip(hash["destination"], mask) : :default,
+        interface: iface,
+        gateway:   build_ip(hash["gateway"], MISSING_VALUE),
+        options:   hash["extrapara"] || ""
+      )
+    end
+
+    # SCR agent for routes files definition
+    def ifroute_term(path)
+      raise ArgumentError if path.nil? || path.empty?
+
+      non_empty_str_term = Yast.term(:String, "^ \t\n")
+      whitespace_term = Yast.term(:Whitespace)
+      optional_whitespace_term = Yast.term(:Optional, whitespace_term)
+      routes_content_term = Yast.term(
+        :List,
+        Yast.term(
+          :Tuple,
+          Yast.term(
+            :destination,
+            non_empty_str_term
+          ),
+          whitespace_term,
+          Yast.term(:gateway, non_empty_str_term),
+          whitespace_term,
+          Yast.term(:netmask, non_empty_str_term),
+          optional_whitespace_term,
+          Yast.term(
+            :Optional,
+            Yast.term(:device, non_empty_str_term)
+          ),
+          optional_whitespace_term,
+          Yast.term(
+            :Optional,
+            Yast.term(
+              :extrapara,
+              Yast.term(:String, "^\n")
+            )
+          )
+        ),
+        "\n"
+      )
```

Comment:
up to this line all is just constant. Only the rest is path based. So maybe having it in constant or at least separated?